app-android/src/androidTest/java/com/ustadmobile/port/android/view/WebChunkFragmentTest.kt

```diff
@@ -0,0 +1,136 @@
+package com.ustadmobile.port.android.view
+
+import android.Manifest
+import androidx.core.os.bundleOf
+import androidx.fragment.app.testing.launchFragmentInContainer
+import androidx.test.espresso.Espresso.onIdle
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.isJavascriptEnabled
+import androidx.test.espresso.web.sugar.Web.onWebView
+import androidx.test.espresso.web.webdriver.DriverAtoms.findElement
+import androidx.test.espresso.web.webdriver.Locator
+import androidx.test.rule.GrantPermissionRule
+import com.toughra.ustadmobile.R
+import com.ustadmobile.adbscreenrecorder.client.AdbScreenRecord
+import com.ustadmobile.adbscreenrecorder.client.AdbScreenRecordRule
+import com.ustadmobile.core.container.ContainerManager
+import com.ustadmobile.core.container.addEntriesFromZipToContainer
+import com.ustadmobile.core.view.UstadView
+import com.ustadmobile.lib.db.entities.Container
+import com.ustadmobile.lib.db.entities.ContentEntry
+import com.ustadmobile.port.sharedse.util.UmFileUtilSe
+import com.ustadmobile.test.core.impl.CrudIdlingResource
+import com.ustadmobile.test.core.impl.DataBindingIdlingResource
+import com.ustadmobile.test.port.android.util.installNavController
+import com.ustadmobile.test.port.android.util.letOnFragment
+import com.ustadmobile.test.rules.ScenarioIdlingResourceRule
+import com.ustadmobile.test.rules.SystemImplTestNavHostRule
+import com.ustadmobile.test.rules.UmAppDatabaseAndroidClientRule
+import com.ustadmobile.test.rules.withScenarioIdlingResourceRule
+import kotlinx.android.synthetic.main.fragment_web_chunk.*
+import org.apache.commons.io.FileUtils
+import org.apache.commons.io.FileUtils.copyInputStreamToFile
+import org.hamcrest.CoreMatchers
+import org.hamcrest.core.AllOf.allOf
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import java.io.File
+
+@AdbScreenRecord("WebChunk Screen Test")
+@ExperimentalStdlibApi
+class WebChunkFragmentTest {
+
+    @JvmField
+    @Rule
+    var dbRule = UmAppDatabaseAndroidClientRule(useDbAsRepo = true)
+
+    @JvmField
+    @Rule
+    var systemImplNavRule = SystemImplTestNavHostRule()
+
+    @JvmField
+    @Rule
+    val dataBindingIdlingResourceRule = ScenarioIdlingResourceRule(DataBindingIdlingResource())
+
+    @JvmField
+    @Rule
+    val screenRecordRule = AdbScreenRecordRule()
+
+    @JvmField
+    @Rule
+    val crudIdlingResourceRule = ScenarioIdlingResourceRule(CrudIdlingResource())
+
+    @get:Rule
+    var permissionRule = GrantPermissionRule.grant(Manifest.permission.WRITE_EXTERNAL_STORAGE,
+            Manifest.permission.READ_EXTERNAL_STORAGE, Manifest.permission.ACCESS_COARSE_LOCATION)
+
+    var container: Container? = null
```

Comment:
For variables that are initialized in the setup block without any conditions, why not use lateinit?